src/targets/cpu/lowering.cpp

```diff
@@ -451,7 +478,22 @@ struct cpu_apply
 
     instruction_ref replace(instruction_ref ins, const operation& op) const
     {
-        return replace(ins, op, ins->inputs());
+        auto inputs = ins->inputs();
+        std::vector<instruction_ref> cont_inputs;
+        for(const auto& i : inputs)
+        {
+            if(not i->get_shape().standard())
+            {
+                auto alloc = insert_allocation(ins, i->get_shape().as_standard());
+                cont_inputs.push_back(
+                    modl->insert_instruction(ins, make_op("dnnl::reorder"), {i, alloc}));
+            }
```

Comment:
This should already be done by `auto_contiguous`. We should probably move it right before lowering like what I did in #3343 for the gpu backend.

Reply:
Yes it should be done except `simplify_reshapes` eliminates it

Reply:
It fails when at `make check` if i move auto_contiguous
```
[   RUN    ] test_scatternd_add

module: "main"
@0 = @literal{4, 3, 1, 7} -> int64_type, {4, 1}, {1, 4}
@1 = allocate[shape=int64_type, {4, 1}, {1, 1},buf_type=nullopt] -> int64_type, {4, 1}, {1, 1}
@2 = contiguous(@0) -> int64_type, {4, 1}, {1, 1}
update = @param:update -> float_type, {4}, {1}
data = @param:data -> float_type, {8}, {1}
@5 = scatternd_add(data,@2,update) -> float_type, {8}, {1}
@6 = @return(@5)


module: "main"
update = @param:update -> float_type, {4}, {1}
@1 = @literal{4, 3, 1, 7} -> int64_type, {1, 4}, {4, 1}
data = @param:data -> float_type, {8}, {1}
@3 = convert[target_type=2](@1) -> float_type, {1, 4}, {4, 1}
@4 = convert[target_type=9](@3) -> int64_type, {1, 4}, {4, 1}
@5 = transpose[permutation={1, 0}](@4) -> int64_type, {4, 1}, {1, 4}
@6 = scatternd_add(data,@5,update) -> float_type, {8}, {1}
@7 = @return(@6)


FAILED: test_scatternd_add
    what(): ../../../../src/targets/cpu/dnnl.cpp:72: to_dnnl_memory_data_type: Unsupported data type

module: "main"
update = @param:update -> float_type, {4}, {1}
@1 = @literal{4, 3, 1, 7} -> int64_type, {1, 4}, {4, 1}
data = @param:data -> float_type, {8}, {1}
@3 = convert[target_type=2](@1) -> float_type, {1, 4}, {4, 1}
@4 = convert[target_type=9](@3) -> int64_type, {1, 4}, {4, 1}
@5 = transpose[permutation={1, 0}](@4) -> int64_type, {4, 1}, {1, 4}
@6 = scatternd_add(data,@5,update) -> float_type, {8}, {1}
@7 = @return(@6)


CMake Error at gdb/test_test_verify_general/run.cmake:16 (message):
  Test failed

```

Reply:
> We should probably move it right before lowering like what I did in https://github.com/ROCm/AMDMIGraphX/pull/3343 for the gpu backend. 

Done